File: src/Car.java
public class Car {
    private final char symbol;
    private Coor start;
    private Coor end;
    private final char direction;

    public Car(char symbol, Coor start, Coor end) {
        this.symbol = symbol;
        this.start = start;
        this.end = end;
        if (start.X == end.X) {
            direction = 'Y';
        } else {
            direction = 'X';
        }
    }

    public void move(int step) {
        if (direction == 'X') {
            start.X += step;
            end.X += step;
        } else {
            start.Y += step;
            end.Y += step;
        }
    }
}

File: src/lib/Car.js
class Car {
  constructor(symbol, start, end) {
    this.symbol = symbol;
    this.start = start;
    this.end = end;
    this.direction = start.X === end.X ? "Y" : "X";
    this.len =
      this.direction === "X"
        ? Math.abs(end.X - start.X)
        : Math.abs(end.Y - start.Y);
  }

  move(step) {
    if (this.direction === "X") {
      this.start.X += step;
      this.end.X += step;
    } else {
      this.start.Y += step;
      this.end.Y += step;
    }
  }
}
